### ISSUE TITLE
Strategies for model training

### ISSUE THREAD

**Issue by DWendou** [ultralytics/ultralytics#19693]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I currently have a dataset specifically for object detection, with all images sized at 1280*960. The training set contains 30,000 images, while the test and validation sets each have 2,000 images. I trained a YOLOv8 rotated object detection model and obtained the corresponding weights best.pt. Over the next month, I will add 10,000 images every week. However, I need to update the model weights every week during this month. How should I iteratively train my model and allocate the gradually added data? Should I add all the data each time for training? How should I adjust the training strategy to make my model better and better? Looking forward to your reply very much！

**Comment by UltralyticsAssistant** [comment]
👋 Hello @DWendou, thank you for sharing your question and for your interest in Ultralytics 🚀! 

We recommend reviewing our [Docs](https://docs.ultralytics.com/) for helpful resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, as they often address common questions related to training strategies and workflows.

If this is a custom training ❓ question, please provide additional details such as dataset examples, training logs, and any adjustments to default parameters. Also, ensure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your iterative training process.

Here are a few steps to get started or troubleshoot:
1. Upgrade to the latest version of `ultralytics` to ensure you have the latest features and fixes:
```bash
pip install -U ultralytics
```
2. Explore YOLO training modes and strategies in different verified environments:
   - **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
   - **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
   - **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
   - **Docker**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

For iterative training, consider strategies like selective fine-tuning, data augmentation, or staged training. Adding newly acquired data incrementally can be effective but requires careful balancing of old and new datasets. Consult the [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) for guidance on optimizing your approach.

If this is a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate.

Join the Ultralytics community for further collaboration:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time discussions
- [Discourse](https://community.ultralytics.com/) for in-depth topics
- [Subreddit](https://reddit.com/r/Ultralytics) for community knowledge sharing

An Ultralytics engineer will review your question soon to provide further assistance. Thank you for contributing to the community 😊!

**Comment by glenn-jocher** [comment]
@DWendou for iterative training with weekly data additions, we recommend these steps:  
1. **Start from latest `best.pt`**: Use your existing weights as pretrained weights for subsequent training sessions via `model = YOLO('path/to/latest.pt')`  
2. **Combine datasets**: Merge new weekly data with existing training data each iteration to maintain pattern diversity  
3. **Adjust training parameters**: Gradually reduce initial learning rate (e.g., `lr0=0.01 → 0.001`) and epochs as the model matures  
4. **Monitor validation**: Keep a fixed validation set to track performance trends  

For detailed implementation, see the Ultralytics guide on handling large datasets and transfer learning in the Model Training Tips documentation.

**Comment by DWendou** [comment]
> [@DWendou](https://github.com/DWendou) for iterative training with weekly data additions, we recommend these steps:
> 
> 1. **Start from latest `best.pt`**: Use your existing weights as pretrained weights for subsequent training sessions via `model = YOLO('path/to/latest.pt')`
> 2. **Combine datasets**: Merge new weekly data with existing training data each iteration to maintain pattern diversity
> 3. **Adjust training parameters**: Gradually reduce initial learning rate (e.g., `lr0=0.01 → 0.001`) and epochs as the model matures
> 4. **Monitor validation**: Keep a fixed validation set to track performance trends
> 
> For detailed implementation, see the Ultralytics guide on handling large datasets and transfer learning in the Model Training Tips documentation.

Thank you very much for your advice, I will try to train my model according to your advice.

**Comment by DWendou** [comment]
![Image](https://github.com/user-attachments/assets/cc631b16-442d-44d5-bd96-71e8b08ee561)I retrained my model on your first three suggestions

**Comment by DWendou** [comment]
I observed that the model achieved its optimal performance at an earlier stage. Last week, my model was trained for 600 epochs before stopping, while this week it stopped after 374 epochs. Additionally, the overall mAP decreased by 0.9%. Could this be due to my adjustment of the initial learning rate (lr0) and the inclusion of last week's best-performing checkpoint (best.pt)? If I were to refrain from adding the best-pt, avoid adjusting lr0, and simply merge the dataset as you suggested in your second point before retraining the model, would the resulting mAP exceed last week's value?

**Comment by glenn-jocher** [comment]
The earlier stopping (374 vs. 600 epochs) suggests your model is converging faster with transfer learning from `best.pt`, which is normal as pretrained weights reduce required training time. The mAP dip could stem from two factors: 

1. **Learning Rate Adjustment**: Overly reduced `lr0` might limit model adaptation to new data patterns. Consider using intermediate values (e.g., `lr0=0.005` instead of 0.001) as a compromise between stability and plasticity.  
2. **Data Distribution Shift**: New weekly data may introduce novel patterns the pretrained model struggles to adapt to immediately.

For your proposed experiment: Training without `best.pt` would reset learned features, likely *reducing* mAP. Instead:  
- Keep using `best.pt` as your starting point  
- Try `lr0=0.005` with `patience=50` (extended convergence window)  
- Validate data balance in merged datasets

These adjustments often help balance stability and adaptability. For detailed parameter tuning strategies, see the Ultralytics guide on [learning rate scheduling](https://docs.ultralytics.com/guides/model-training-tips/#learning-rate) and [early stopping configuration](https://docs.ultralytics.com/usage/cfg/#train-settings).

**Comment by DWendou** [comment]
> The earlier stopping (374 vs. 600 epochs) suggests your model is converging faster with transfer learning from `best.pt`, which is normal as pretrained weights reduce required training time. The mAP dip could stem from two factors:
> 
> 1. **Learning Rate Adjustment**: Overly reduced `lr0` might limit model adaptation to new data patterns. Consider using intermediate values (e.g., `lr0=0.005` instead of 0.001) as a compromise between stability and plasticity.
> 2. **Data Distribution Shift**: New weekly data may introduce novel patterns the pretrained model struggles to adapt to immediately.
> 
> For your proposed experiment: Training without `best.pt` would reset learned features, likely _reducing_ mAP. Instead:
> 
> * Keep using `best.pt` as your starting point
> * Try `lr0=0.005` with `patience=50` (extended convergence window)
> * Validate data balance in merged datasets
> 
> These adjustments often help balance stability and adaptability. For detailed parameter tuning strategies, see the Ultralytics guide on [learning rate scheduling](https://docs.ultralytics.com/guides/model-training-tips/#learning-rate) and [early stopping configuration](https://docs.ultralytics.com/usage/cfg/#train-settings).

Thanks for your advice, I have already had a good result, I will carefully learn a parameter tuning strategies。

**Comment by glenn-jocher** [comment]
Great to hear you're seeing positive results! For deeper insights into parameter tuning, explore our [Learning Rate Scheduling guide](https://docs.ultralytics.com/guides/model-training-tips/#learning-rate) and [Early Stopping Configuration docs](https://docs.ultralytics.com/usage/cfg/#train-settings). Happy training! 🚀
